### PR TITLE
Added `Vault` component and stories

### LINKED ISSUE #435: Create `VaultLiquidation` component
### Goal
Introduce a new component that display the basic vault information. You can use the component `/components/auctions/collateral/CollateralAuction` as a base on which you can build this new component. Please also take the "See additional information" button to hide the last 4 rows of the table. 

### Assets
Screenshot 2022-08-30 at 13 50 01

### PR DESCRIPTION
**Pull request by zoey-kaiser**
Closes #435 

Screenshot 2022-09-12 at 16 13 58

Screenshot 2022-09-09 at 16 25 47

Checklist:
- [X] issue number linked above after pound (`#`)
    - replace "Closes " with "Contributes to" or other if this PR does not close the issue
- [X] issue checkboxes are all addressed
- [X] manually checked my feature / not applicable
- [X] wrote tests / not applicable
- [X] attached screenshots / not applicable

### CODE REVIEW

File: frontend/components/vault/Vault.vue [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+                    <template v-if="!error">
+                        The vault
+                        <format-address type="address" :value="vaultTransaction.address" shorten disable /> contains
+                        <format-currency
+                            :value="vaultTransaction.collateralAmount"
+                            :currency="vaultTransaction.collateralType"
+                        />. The initial debt taken by the vault is
+                        <format-currency :value="vaultTransaction.initialDebtDai" currency="DAI" />. The current
+                        proximity to liquidation is
+                        <AnimatedNumber :value="vaultTransaction.proximityToLiquidation" />. The next price update will
```
Comment (valiafetisov) [inline frontend/components/vault/Vault.vue:120]: How many symbols will `AnimatedNumber` display after the floating point by default or all of them?

File: frontend/components/vault/Vault.vue [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+                        </nuxt-link>
+                    </div>
+                </div>
+            </Alert>
+            <TextBlock class="mt-4">
+                This vault was liquidated <TimeTill :date="vaultTransaction.liqudiationDate" /> in the transaction
+                <FormatAddress :value="vaultTransaction.transactionHash" />.
+            </TextBlock>
+        </div>
+        <Loading v-else-if="areVaultsFetching" is-loading class="w-full self-center Loading h-48" />
```
Comment (valiafetisov) [inline frontend/components/vault/Vault.vue:153]: What happens on `v-else`? Does page stays empty?

File: frontend/components/vault/Vault.vue [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+                };
+            } else if (!this.areVaultsFetching && this.vaultTransaction.state === 'liquidated') {
+                return {
+                    error: 'This vault has already been liquidated',
+                    showBanner: false,
+                };
+            }
+            return null;
+        },
+        vaultLink(): string | undefined {
```
Comment (valiafetisov) [inline frontend/components/vault/Vault.vue:236]:
```suggestion
        auctionLink(): string | undefined {
```

File: frontend/components/vault/Vault.vue [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+            if (!this.vaultTransaction || !this.vaultTransaction.auctionId) {
+                return undefined;
+            }
+            const link = generateLink(this.network, 'collateral');
+            const formattedAuctionId = this.vaultTransaction.auctionId.replace(':', '%3A');
```
Comment (valiafetisov) [inline frontend/components/vault/Vault.vue:241]: As commented in another PR, please don't reinvent `encodeURIComponent`. You might want to create a helper for that

File: frontend/components/vault/Vault.vue [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+    watch: {
+        isExplanationsShown: {
+            immediate: true,
+            handler(newIsExplanationsShown) {
+                if (!newIsExplanationsShown) {
+                    this.isTableExpanded = true;
+                }
+            },
+        },
+    },
```
Comment (valiafetisov) [inline frontend/components/vault/Vault.vue:255]: Is this used?

File: frontend/components/vault/Vault.vue (hunk 2 above)
Comment (zoey-kaiser) [reply frontend/components/vault/Vault.vue:153]: Nothing will be displayed. This is the same logic we used in all the Auction Components too, so I took it over:

https://github.com/sidestream-tech/unified-auctions-ui/blob/main/frontend/components/auction/collateral/CollateralAuction.vue#L262

File: frontend/components/vault/Vault.vue (hunk 5 above)
Comment (zoey-kaiser) [reply frontend/components/vault/Vault.vue:255]: This again is from the other Auction components:

https://github.com/sidestream-tech/unified-auctions-ui/blob/main/frontend/components/auction/collateral/CollateralAuction.vue#L388

We do use the functionality here to expand the table, so I assume it is needed, if we need it in the other ones.

File: frontend/components/vault/Vault.vue (hunk 1 above)
Comment (valiafetisov) [reply frontend/components/vault/Vault.vue:120]: > by default 

By default. In other words, if `0.123456789` is provided what will be displayed?

**Review by valiafetisov — APPROVED** [review]

File: frontend/components/vault/Vault.vue (hunk 1 above)
Comment (zoey-kaiser) [reply frontend/components/vault/Vault.vue:120]: It displays the first number after the decimal point, that is not zero and the preceding number. Therefore:

`0.12`

**Pull request merged**